lib/std/RingBuffer.zig

```diff
@@ -0,0 +1,136 @@
+//! This ring buffer stores read and write indices while being able to utilise
+//! the full backing slice by incrementing the indices modulo twice the slice's
+//! length and reducing indices modulo the slice's length on slice access. This
+//! means that whether the ring buffer if full or empty can be distinguished by
+//! looking at the difference between the read and write indices without adding
+//! an extra boolean flag or having to reserve a slot in the buffer.
+//!
+//! This ring buffer has not been implemented with thread safety in mind, and
+//! therefore should not be assumed to be suitable for use cases involving
+//! separate reader and writer threads.
+
+const Allocator = @import("std").mem.Allocator;
+const assert = @import("std").debug.assert;
+
+const RingBuffer = @This();
```

Comment:
This API has quite a bit of overlap with `std.fifo.LinearFifo()`, I'm not sure we should have both in the standard library.

Reply:
To add to this discussion, it looks like the lzma implementation has its own `LzCircularBuffer` type as well which is also a somewhat specialized ring buffer.

Reply:
Then maybe `LzCircularBuffer` should be replaced by this `RingBuffer`, too.

Reply:
Opened #19231 to track this